goal_src/jak2/engine/game/idle-control.gc

```diff
@@ -30,6 +30,98 @@
   :flag-assert         #xb00000010
   (:methods
     (idle-control-method-9 (_type_ (pointer uint32)) none 9)
-    (idle-control-method-10 (_type_ process-drawable) none 10)
+    (idle-control-method-10 (_type_ process-drawable) none :behavior process-drawable 10)
     )
   )
+
+
+;; WARN: Return type mismatch idle-control vs none.
+(defmethod idle-control-method-9 idle-control ((obj idle-control) (arg0 (pointer uint32)))
+  (set! (-> obj anim) arg0)
+  (set! (-> obj current) arg0)
+  (set! (-> obj counter) 0)
+  (set! (-> obj target) 0)
+  (none)
+  )
+
+;; TODO - method manually patched (maybe incorrectly) around `self` changing
+;; WARN: Function (method 10 idle-control) has a return type of none, but the expression builder found a return statement.
+(defmethod idle-control-method-10 idle-control ((obj idle-control) (arg0 process-drawable))
+  (local-vars (a1-1 int))
+  (when (nonzero? (-> obj anim))
+    (let ((s5-0 self))
+      ;; (set! self arg0)
```

Comment:
I think we need to change this to have the correct process pointer.
You could wrap this whole thing in a 
```
(with-pp arg0
  <all the stuff>
  )
  ```
  and that should do the right thing.